### ISSUE TITLE
Index: Support for arbitraty encoding

### ISSUE THREAD

**Issue by hy26v** [TabbyML/tabby#1466]
We are using an old framework that has the need to be encoded in windows-1252 thus all frontend components that utilize parts of this framework have to be encoded as such and it is still being used at some points. I assume there will be other people using legacy code bases where encodings other than UTF-8 are being utilized. Please add support for recognizing the encoding of  a file before indexing it in said encoding.

Also see https://github.com/TabbyML/tabby/issues/1287 (someone with the same issue apparently)

**Comment by wsxiaoys** [comment]
cc @boxbeam for comments, if we wanna support other encodings, does that mean we need run something like iconv?

**Comment by boxbeam** [comment]
Yes, a way to convert at time of reading would probably be our best bet.
